### PR TITLE
Security fix cherry-picks for 2021.3.2

### PR DESCRIPTION
**Pull request by seanbudd**
This cherry-picks the following commits to rc, for a 2021.3.2 release:

- c29fa5646a8b1ad240345842e1ed858218393e83
- 7afc3ed579b6123ffe3ebc86140f131e03ebaedc
- 9525b63366b7b62827827823a9354884c3c0e2f4

To ensure the PR builds this commit was cherry-picked:
- dd17adae5ee5a63344d1b65914861ee442376f11

A changelog section has been added for 2021.3.2

This should be merged as a merge commit, tested and then turned into a 2021.3.2 release.

Advisories will be included in the release post, with links to related GitHub discussions.

### CODE REVIEW

**Review by michaelDCurran — DISMISSED** [review]

**Comment by cary-rowen** [comment]
I am interested in this security fix, I want to download the tryBuild for testing, trying to find more security risks, but at present it has exceeded the limit of 1024MB and cannot be downloaded

**Comment by Brian1Gaff** [comment]
Yes, why do they have this limit all of a sudden? I would have thought a 
development sitewouldneed more bandwidth, not less.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

File: source/globalVars.py [hunk 1]
```diff
@@ -1,35 +1,61 @@
...
+
+import argparse
+import os
+import typing
+
+if typing.TYPE_CHECKING:
+	import NVDAObjects  # noqa: F401 used for type checking only
+
+
+class DefautAppArgs(argparse.Namespace):
```
Comment (lukaszgo1) [inline source/globalVars.py:25]: I don't think introducing this code into RC is a good idea. Providing the dummy implementation of `appArgs` causes #13094 so to fix this you would also need to backport #13099 which seems an unnecessary risk.

Comment (michaelDCurran) [reply source/globalVars.py:25]: I'm assuming this was not intentional, and that it was accidentally pulled in due to a bad merge conflict fix. 
This code should definitely be removed.

**Review by michaelDCurran — CHANGES_REQUESTED** [review]

File: source/globalVars.py (hunk 1 above)
Comment (seanbudd) [reply source/globalVars.py]: Correct. I've rebased and fixed the cherry-pick.

I've also removed the files with changes that consist of unnecessary typing additions.

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**